src/app/shared/element-child-data-classes-child-elements-list/flattened-data-classes-list.component.ts

```diff
@@ -0,0 +1,241 @@
+/*
+Copyright 2020-2024 University of Oxford and NHS England
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+import {
+  Component,
+  Input,
+  ViewChildren,
+  ViewChild,
+  AfterViewInit,
+  ElementRef,
+  EventEmitter,
+  ChangeDetectorRef,
+  OnInit,
+  Output
+} from '@angular/core';
+import { StateHandlerService } from '@mdm/services/handlers/state-handler.service';
+import { MdmResourcesService } from '@mdm/modules/resources';
+import { merge, Observable } from 'rxjs';
+import { catchError, map, startWith, switchMap } from 'rxjs/operators';
+import { MatSort } from '@angular/material/sort';
+import { MdmPaginatorComponent } from '../mdm-paginator/mdm-paginator';
+import { MatDialog } from '@angular/material/dialog';
+import { BulkDeleteModalComponent } from '@mdm/modals/bulk-delete-modal/bulk-delete-modal.component';
+import { GridService } from '@mdm/services/grid.service';
+import { CdkDragDrop, moveItemInArray } from '@angular/cdk/drag-drop';
+import { MatTable } from '@angular/material/table';
+import { MessageHandlerService } from '@mdm/services';
+import { DataClass } from '@maurodatamapper/mdm-resources';
+
+@Component({
+  selector: 'mdm-flattened-data-classes-list',
+  templateUrl: './flattened-data-classes-list.component.html',
+  styleUrls: ['./flattened-data-classes-list.component.sass']
+})
+export class FlattenedDataClassesComponent implements AfterViewInit, OnInit {
+  @Input() parentDataModel: any;
```

Comment:
What are these `any` types meant to be? Without type information, this code becomes harder to interpret and the code editor is not going to help catch bugs for you. Please add some type information wherever possible.